annotations/utils.py

```diff
@@ -34,3 +34,36 @@ def basepath(request):
     else:
         scheme = 'http://'
     return scheme + request.get_host() + settings.SUBPATH
+
+
+def get_pagination_metadata(total_items, page, items_per_page):
+    """
+    Calculate pagination metadata such as total pages, page range, and the current page.
+    
+    Args:
+        total_items (int): Total number of items.
+        page (int): The current page number requested.
+        items_per_page (int): Number of items per page.
+    
+    Returns:
+        dict: Pagination metadata including total pages, page range, and the validated current page.
+    """
+
+    # Calculate total pages by dividing total items by items per page. If there are any leftover items, an additional page is created.
+    total_pages = (total_items + items_per_page - 1) // items_per_page
```

Comment:
I don't understand that, why does this take the total number of items and adds another page to ?

Reply:
The problem is if there are leftover items that don’t perfectly fit into the available pages.

100 items, 20 items per page: 100 // 20 = 5
101 items, 20 items per page: 100// 20 = 5 (There should be 6 here but we miss one due to floor division)

`(total_items + items_per_page - 1) // items_per_page` solves this issue by forcing the integer division to round up whenever there are leftover items.

(101+20−1)//20=120÷20=6 pages

Reply:
why not just do `math.ceil(total_items/items_per_page)`? with a comment about rounding up to accommodate "incomplete"  last pages. a lot easier to understand.

Reply:
it totally slipped my mind that math.ceil does exactly that 🤦‍♂️,  I'll make the changes.